enterprise/frontend/src/embedding-sdk/store/auth/auth.ts

```diff
@@ -38,27 +38,19 @@ export const initAuth = createAsyncThunk(
       // API key setup
       api.apiKey = authConfig.apiKey;
     } else if (isValidInstanceUrl) {
-      try {
-        // SSO setup
-        api.onBeforeRequest = async () => {
-          const session = await dispatch(
-            getOrRefreshSession(authConfig.metabaseInstanceUrl),
-          ).unwrap();
-          if (session?.id) {
-            api.sessionToken = session.id;
-          }
-        };
-        // verify that the session is actually valid before proceeding
-        await dispatch(
+      // SSO setup
+      api.onBeforeRequest = async () => {
+        const session = await dispatch(
           getOrRefreshSession(authConfig.metabaseInstanceUrl),
         ).unwrap();
-      } catch (e) {
-        // TODO: make this more specific to whatever status code we receive:
-        // this is good for a 400
-        throw new Error(
-          `Unable to connect to instance at ${authConfig.metabaseInstanceUrl}`,
-        );
```

Comment:
Do we have custom error handling in `getOrRefreshSession` that can replace this `throw new Error("Unable to connect to instance")`? Since `unwrap` can throw an error.